internal/systemd/systemd.go

```diff
@@ -0,0 +1,174 @@
+package systemd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	systemd "github.com/coreos/go-systemd/v22/dbus"
+	"github.com/subpop/go-log"
+)
+
+type ConnectionType int
+
+const (
+	ConnectionTypeSystem ConnectionType = iota
+	ConnectionTypeUser
+)
+
+type Conn struct {
+	ctx  context.Context
+	conn *systemd.Conn
+}
+
+// NewConnectionContext creates a new connection to the given systemd service.
+func NewConnectionContext(ctx context.Context, connectionType ConnectionType) (*Conn, error) {
+	var conn *systemd.Conn
+	var err error
+	if connectionType == ConnectionTypeSystem {
+		conn, err = systemd.NewSystemConnectionContext(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("cannot establish system connection to systemd: %v", err)
+		}
+	} else {
+		conn, err = systemd.NewUserConnectionContext(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("cannot establish user connection to systemd: %v", err)
+		}
+	}
+
+	return &Conn{
+		ctx:  ctx,
+		conn: conn,
+	}, nil
+}
+
+func (c *Conn) Close() {
+	c.conn.Close()
+}
+
+// Reload instructs systemd to scan for and reload unit files.
+func (c *Conn) Reload() error {
+	return c.conn.ReloadContext(c.ctx)
+}
+
+// EnableUnit enables the named unit. If activate is true, it also starts the
+// unit. If runtime is true, the unit is enabled for the runtime only (/run). If
+// false, it is enabled persistently (/etc).
+func (c *Conn) EnableUnit(name string, activate bool, runtime bool) error {
```

Comment:
The name of the method could be misleading. What about `EnableActivateUnit()` ... Well, I don't like this proposed name, but I don't have better idea.

Reply:
I feel like having `activate` as an argument parameter is pretty clear. I think this is the Go convention too. I want to avoid the Objective-C pattern of embedded every argument parameter name in the method name: `enableUnit:name:activate:runtime`.

Reply:
I'm not strongly convinced that this method signature is unclear. The brevity of
the base function name, plus the names of the argument parameters, seem pretty
clear to me.